src/app/file-manager/transferables-grid/transferables-grid.component.ts

```diff
@@ -1,56 +1,94 @@
+import { ExportToS3State } from '@app/file-manager/reducers/export-to-s3.reducer';
+
+import { MatDialog } from '@angular/material';
 import 'rxjs/add/operator/startWith';
 import 'rxjs/add/observable/forkJoin';
 import 'rxjs/add/operator/map';
 import 'rxjs/add/observable/merge';
 import { GcsService } from '@app/file-manager/services/gcs.service';
-import { StatusService } from '../services/status.service';
-import { AppState } from '../dbstate/app-state';
-import { FilesDatabase } from '../dbstate/files-database';
-import { LimitTransferablesService } from '../services/limit-transferables.service';
+import { StatusService } from '@app/file-manager/services/status.service';
+
+import { LimitTransferablesService } from '@app/file-manager/services/limit-transferables.service';
 import { Type } from '@app/file-manager/models/type';
 import { ItemStatus } from '@app/file-manager/models/item-status';
 import { S3ExportService } from '@app/file-manager/services/s3-export.service';
 import { NgxSpinnerService } from 'ngx-spinner';
-import { WarningModalComponent } from '../warning-modal/warning-modal.component';
+import { ProgressBar } from 'primeng/primeng';
+import { Observable } from 'rxjs/Observable';
+import { WarningModalComponent } from '@app/file-manager/warning-modal/warning-modal.component';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 @Component({
   selector: 'app-transferables-grid',
   templateUrl: './transferables-grid.component.html',
-  styleUrls: ['./transferables-grid.component.css']
+  styleUrls: ['./transferables-grid.component.css'],
+  encapsulation: ViewEncapsulation.None
 })
 
 export class TransferablesGridComponent implements OnInit, AfterViewInit {
 
-  /**
-   * These values are not persisting through the app execution.
-   * When receiving info from any observable we get a new instance from the component,
-   * which redefines any local variable. In this case, these values represent changes in the UI.
-   * Todo : put needed properties on an object that has the correct lifetime. The Store is the best place.
-   * */
-  static isExporting: Boolean = false;
-  static firstIteration: Boolean = true;
-  @ViewChild(MatPaginator) paginator: MatPaginator;
-  @ViewChild(MatSort) sort: MatSort;
-
+  tabFilter = '';
+  selectedIndex: number;
   displayedColumns = ['name', 'size', 'status', 'progress', 'actions'];
-  dataSource = new MatTableDataSource([]);
-  filesDatabase: FilesDatabase;
-  generalProgress = 0;
-  generalUploadProgress = 0;
-  uploadInProgress = false;
-  downloadInProgress = false;
-  disabledDownload = false;
-  disabledUpload = false;
-  exportToS3InProgress = false;
-  exportToGcpItems = new FilesDatabase(this.store);
-  exportToS3Canceled = false;
 
+  // ---------------------------- progress info from here ----------------
+  downCompleted = 0;
+  downTotal = 0;
+  downProgress = 0;
+  downPending = false;
+  downInProgress = false;
+  downCanceled = false;
+  downFinished = false;
```

Comment:
Using a class for each case would be helpful here as well.

Reply:
yes, we know there are code to refactor here. We'll nedd to re-test after this change, will not be ready until Monday.